rpcs3/Emu/Cell/lv2/sys_fs.cpp

```diff
@@ -240,39 +240,39 @@ error_code sys_fs_test(ppu_thread& ppu, u32 arg1, u32 arg2, vm::ptr<u32> arg3, u
 	return CELL_OK;
 }
 
-error_code sys_fs_open(ppu_thread& ppu, vm::cptr<char> path, s32 flags, vm::ptr<u32> fd, s32 mode, vm::cptr<void> arg, u64 size)
+std::pair<CellError, fs::file> lv2_file::open(std::string_view path, s32 flags, s32 mode, const void* arg, u64 size, std::string* processed_path)
```

Comment:
If you return `std::tuple<CellError, std::string, fs::file` in such order, you won't need to change all the return statements.

Reply:
Oh, you need an aggregate for it. Can make a custom return type with 3 fields.